docs/io/misc.rst

```diff
@@ -22,246 +22,3 @@ listed in the `astropy.io.misc` section.
 
 .. automodapi:: astropy.io.misc.parquet
    :headings: =-
-
-astropy.io.misc.asdf Package
```

Comment:
Instead of outright deleting this section, maybe at least keep the title but point to your new page and put a warning here that this subpackage has been deprecated?

After all, it is not removed yet, just deprecated.